### PR TITLE
Replace `javax.inject.*`, `javax.annotations.Nullable`, `javax.annotations.Nonnull` with Jakarta annotations

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: core/src/main/java/io/micronaut/core/annotation/AnnotatedElement.java [hunk 1]
```diff
@@ -29,27 +29,27 @@ public interface AnnotatedElement extends AnnotationMetadataProvider, Named {
      * @return Whether the element is nullable.
      */
     default boolean isDeclaredNullable() {
-        return getAnnotationMetadata().hasDeclaredStereotype(AnnotationUtil.NULLABLE);
+        return getAnnotationMetadata().hasDeclaredStereotype(AnnotationUtil.NULLABLE) || getAnnotationMetadata().hasDeclaredStereotype(AnnotationUtil.JAVAX_NULLABLE);
```
Comment (graemerocher) [inline core/src/main/java/io/micronaut/core/annotation/AnnotatedElement.java:32]: why is this needed? The annotation should be mapped

**Review by graemerocher — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

File: core/src/main/java/io/micronaut/core/annotation/AnnotatedElement.java (hunk 1 above)
Comment (dstepanov) [reply core/src/main/java/io/micronaut/core/annotation/AnnotatedElement.java:32]: It was mapped to the javax one so everything compiled before would have the old annotation. We need to recompile the platform
